addons/event/models/event_event.py

```diff
@@ -455,6 +455,11 @@ def _compute_event_ticket_ids(self):
                 ]
             event.event_ticket_ids = command
 
+    @api.onchange('stage_id')
+    def _onchange_stage_id(self):
+        for event in self:
+            event.kanban_state = 'normal'
```

Comment:
As a general rule, we try to avoid implementing behaviors with the "onchange" method, it tends to be deprecated over time.

Instead, I see in projets that the "write" method is overridden to handle the kanban_state reset when the task changes stage:

```py
if 'stage_id' in vals:
    ...
    # reset kanban state when changing stage
    if 'kanban_state' not in vals:
        vals['kanban_state'] = 'normal'
```